indico/modules/events/persons/schemas.py

```diff
@@ -5,17 +5,61 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import fields
+from marshmallow import EXCLUDE, fields, post_dump
 
 from indico.core.marshmallow import mm
 from indico.modules.events.models.persons import EventPerson
+from indico.modules.users.models.users import UserTitle
+from indico.util.i18n import orig_string
+
+
+class PersonLinkSchema(mm.Schema):
+    class Meta:
+        unknown = EXCLUDE
+
+    _type = fields.Constant('PersonLink')
+    person_id = fields.Int()
+    user_id = fields.Int(attribute='person.user_id')
+    user_identifier = fields.String(attribute='user.identifier')
+    first_name = fields.String(load_default='')
+    last_name = fields.String(required=True)
+    title = fields.Method('get_title', deserialize='load_title')
+    affiliation = fields.String(load_default='')
+    phone = fields.String(load_default='')
+    address = fields.String(load_default='')
+    email = fields.String(required=True)
+    display_order = fields.Int(load_default=0, dump_default=0)
+    avatar_url = fields.Function(lambda o: o.person.user.avatar_url if o.person.user else None)
+
+    def get_title(self, obj):
+        return obj.title
```

Comment:
I think this will break editing when you use a language where the titles are translated to something else (since this one will return the translated version, and that won't match any of the dropdown options client-side)